.build/build-sonar.xml

```diff
@@ -0,0 +1,167 @@
+        <check-server-status url="${sonar.host.url}"/>
+    </target>
+
+    <target name="maybe-set-sonar-server-url" depends="sonar-init" if="env.SONAR_SERVER_URL">
+        <property name="sonar.host.url" value="${env.SONAR_SERVER_URL}"/>
+        <fail unless="env.SONAR_CASSANDRA_TOKEN" message="SONAR_CASSANDRA_TOKEN environment variable must be set when using Sonar server defined in env SONAR_SERVER_URL"/>
+        <property name="sonar.token" value="${env.SONAR_CASSANDRA_TOKEN}"/>
+    </target>
+
+    <target name="maybe-start-sonar-server" depends="sonar-init" unless="env.SONAR_SERVER_URL">
+        <property name="sonar.host.url" value="http://127.0.0.1:9000"/>
+        <local name="sonar-server.running"/>
+        <condition property="sonar-server.running" value="true">
+            <socket server="127.0.0.1" port="9000"/>
+        </condition>
+        <antcall target="start-sonarqube" unless:set="sonar-server.running" inheritrefs="true"/>
+
+        <exec executable="bash" failifexecutionfails="false" failonerror="true" logerror="false">
+            <arg line="-c"/>
+            <arg value="curl -s -u admin:password -X POST '${sonar.host.url}/api/user_tokens/revoke?name=cassandra'"/>
+        </exec>
+
+        <exec executable="bash" failifexecutionfails="true" failonerror="true" logerror="false">
+            <arg line="-c"/>
+            <arg value="curl -s -u admin:password -X POST '${sonar.host.url}/api/user_tokens/generate?name=cassandra&amp;type=GLOBAL_ANALYSIS_TOKEN' | jq -r '.token'"/>
+            <redirector outputproperty="sonar.token"/>
+        </exec>
+    </target>
+
+    <target name="start-sonarqube" depends="sonar-init">
+        <exec executable="docker" dir="${sonar.workdir}" failifexecutionfails="true" failonerror="true" logerror="true">
+            <arg value="run"/>
+            <arg value="-d"/>
+            <arg value="--name"/>
+            <arg value="sonarqube"/>
+            <arg value="-e"/>
+            <arg value="SONAR_ES_BOOTSTRAP_CHECKS_DISABLE=true"/>
+            <arg value="-p"/>
+            <arg value="9000:9000"/>
+            <arg value="-v"/>
+            <arg value="${sonar.workdir}/data:/opt/sonarqube/data"/>
+            <arg value="-v"/>
+            <arg value="${sonar.workdir}/logs:/opt/sonarqube/logs"/>
+            <arg value="-v"/>
+            <arg value="${sonar.workdir}/extensions:/opt/sonarqube/extensions"/>
+            <arg value="sonarqube:community"/>
+        </exec>
+
+        <retry retrycount="60" retrydelay="1000">
+            <check-server-status url="${sonar.host.url}"/>
+        </retry>
+
+        <exec executable="curl" failifexecutionfails="true" failonerror="true" logerror="false">
+            <arg line="-u admin:admin"/>
+            <arg line="-X POST"/>
+            <arg line="-s"/>
+            <arg value="${sonar.host.url}/api/users/change_password?login=admin&amp;password=password&amp;previousPassword=admin"/>
```

Comment:
What is wrong with the `admin:admin`? Why do we need to change it? This is the requirement of the sonar itself, right?

Reply:
Yes, it forces the user to change the password as the first thing after startup. It does not allow for using `admin` as a password again.